app/controllers/vendors_controller.rb

```diff
@@ -1,2 +1,54 @@
 class VendorsController < ApplicationController
+  def index
+    @vendors = Vendor.all
+  end
+
+  def new
+    @vendor = Vendor.new
+  end
+
+  def create
+    Vendor.create(vendor_params[:vendor])
+    redirect_to "/vendors"
+  end
+
+  def edit
+    id = params[:id]
+    @vendor = Vendor.find(id)
+  end
+
+  def show
+    id = params[:id]
+    if session[:vendor_id] != id
+      render 'vendors/error'
+    else
+      @vendor = Vendor.find(id)
+      @products = Vendor.find(id).products
+      @sales = Vendor.find(id).sales
```

Comment:
By calling the `find` multiple times you are querying the database each time the `find` is called. You should be using the `@vendor` variable rather than retrieving the vendor each individual time.